add example: construct guest resource

DougAnderson444 opened the pull request:
### Construct guest resource example
- uses `Value::Bool(false)` as results placeholder
- reverted `pub` to `pub(crate)` as no it's not necessary

Closes https://github.com/DouglasDwyer/wasm_component_layer/issues/23

DouglasDwyer commented on examples/guest_resource/.vscode/settings.json:
```diff
@@ -0,0 +1,10 @@
+{
```
Since everyone uses different IDEs, I suggest removing `vscode` files or adding them to the `.gitignore`.

DougAnderson444 replied:
Agreed. I'm on neovim personally, so it doesn't even apply to me!

Again, it's a `cargo-component` template inclusion, which is what I've been using to build new components.

I'll remove it

DouglasDwyer reviewed (commented):
Thanks for creating this example - I had a few small questions/comments on PR structure, but after they are addressed I will merge it :)

DougAnderson444 commented on rust-toolchain.toml:
```diff
@@ -0,0 +1,2 @@
+[toolchain]
```
Originally I got a build error because [per-package-target](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#per-package-target) needs nightly and I had stable as the default.

DougAnderson444 replied:
I set it to `stable`, still compiling fine, so I will remove it too. Might have been another `cargo-component` related issue?

DougAnderson444 replied:
It appears to be the `package.default-target` and `package.forced-target` manifest keys that demand nightly feature flags:

This appeared when I added in `forced-target = "wasm32-unknown-unknown"`: 
![image](https://github.com/user-attachments/assets/c30a60e2-36fb-470c-8ba1-463817848a2c)

DougAnderson444 replied:
My mistake, I read that wrong. It's a _cargo_ nightly feature, not a _rustc_ nightly feature.

I needed to add:

```toml
cargo-features = ["per-package-target"]

forced-target = "wasm32-unknown-unknown"
```

DougAnderson444 commented on examples/guest_resource/src/bindings.rs:
```diff
@@ -0,0 +1,440 @@
+#[allow(dead_code)]
```
It's the latest way of doing things with `wit-bindgen-rt`, as per the `cargo-component` template.

But that said, I did just try it as the `wit-bindgen` way, and that works too. If you prefer the examples to be consistent with each other, we can go with: 

```rust
wit_bindgen::generate!({
    path: "wit/world.wit"
});

export!(Component);

//..
```

I'm not fussed either way, happy to commit ^^^ those change.

DougAnderson444 replied:
Note that I used the latest version of `wit-bindgen` so the api is a bit different, but same effect.

DouglasDwyer replied:
Okay, let's stick with this because it is simpler and `wit_bindgen` also appears to be well-supported. Users concerned about compile times and dependencies can use the `bindings.rs` solution in their own projects. For an example it's better to be simpler :)

DougAnderson444 commented:
@DouglasDwyer thanks for the feedback! I addressed your issues, committed them, cleaned up a couple other dead code lines, and even learned a few new things myself along the way.

If you spot anything else pls let me know.

DouglasDwyer commented:
Thanks for your contribution!
